zkevm-circuits/src/evm_circuit/util/math_gadget/abs_word.rs

```diff
@@ -33,8 +34,10 @@ impl<F: Field> AbsWordGadget<F> {
         let x = cb.query_word32();
         let x_abs = cb.query_word32();
         let sum = cb.query_word32();
-        let (x_lo, x_hi) = x.to_word().to_lo_hi();
-        let (x_abs_lo, x_abs_hi) = x_abs.to_word().to_lo_hi();
+        let binding = x.to_word();
+        let (x_lo, x_hi) = binding.to_lo_hi();
+        let binding = x_abs.to_word();
+        let (x_abs_lo, x_abs_hi) = binding.to_lo_hi();
```

Comment:
I'm keep getting this rust compiler complains for me to add these `binding` variables.

```
temporary value dropped while borrowed
consider using a `let` binding to create a longer lived value 
```

Reply:
I see that both `to_word` returns an owned type, but `to_lo_hi` returns a borrowed type, so the result of `to_word` needs to be alive for the result of `to_lo_hi` to be available.

I think an issue here is the naming.  `to_*` methods should not return a borrowed type (for that case there's `as_*` methods).  Possible solutions are:
- Rename `to_lo_hi` to `as_lo_hi` to make the borrowed result clear, and then perhaps use some clone helper methods so that we can do `x.to_word().as_lo_hi().cloned()`
- Implement `to_lo_hi` to return an owned result.

Reply:
I went with option 2, "Implement to_lo_hi to return an owned result."

[31be90a](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1435/commits/31be90a97d6d528cb7e00014d2869ea5b11e0314)